### PR TITLE
Add ability to upload an SBOM to container registries

### PR DESCRIPTION
**Pull request by Brend-Smits**
This pull request adds a new step in the workflow that enables the ability to upload a signed SBOM into the container registries. It does this with the help of the Cosign tooling.
It has been tested here: https://github.com/philips-labs/slsa-provenance-action/runs/5063547912?check_suite_focus=true
With the artifact stored here: https://github.com/philips-labs/slsa-provenance-action/pkgs/container/slsa-provenance%2Fsbom.spdx

The SBOM that is uploaded to the registry is a sample SBOM that should not be used. We need a new pull request to add the generation of an SBOM.

To download the contents of the artifact I used a tool called [sget](https://github.com/sigstore/cosign#sget) (from Cosign):
``sget --key cosign.pub ghcr.io/philips-labs/slsa-provenance/sbom.spdx:v0.6.1-draft``

If this is the desired behavior and the pull request is good to merge, please wait with merging until we add a new step that also generates a valid SBOM for this project.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -139,6 +139,66 @@ jobs:
...
+          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u philipssoftware --password-stdin
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login -u ${{ github.actor }} --password-stdin ghcr.io
+
+      - name: Install signing key
+        run: |
+          echo '${{ secrets.COSIGN_PRIVATE_KEY }}' > cosign.key
+
+      - name: Upload SBOM to OCI
+        run: |
+          cosign upload blob -f test-data/sbom.spdx ${{ matrix.repo }}/sbom.spdx@${{ needs.release.outputs.container_digest }}
```
Comment (marcofranssen) [inline .github/workflows/ci.yaml:179]:
```suggestion
          cosign upload blob -f test-data/sbom.spdx ${{ matrix.repo }}/sboms@${{ needs.release.outputs.container_digest }}
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -139,6 +139,66 @@ jobs:
...
+
+      - name: Sign the SBOM
+        run: |
+          cosign sign --key cosign.key ${{ matrix.repo }}/sbom.spdx@${{ needs.release.outputs.container_digest }}
```
Comment (marcofranssen) [inline .github/workflows/ci.yaml:183]:
```suggestion
          cosign upload blob -f test-data/sbom.spdx ${{ matrix.repo }}/sboms@${{ needs.release.outputs.container_digest }}
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -139,6 +139,66 @@ jobs:
...
+        env:
+          COSIGN_PASSWORD: ${{ secrets.COSIGN_PASSWORD }}
+
+      - name: Verify the signature of the SBOM
+        run: |
+          cosign verify --key cosign.pub ${{ matrix.repo }}/sbom.spdx@${{ needs.release.outputs.container_digest }}
```
Comment (marcofranssen) [inline .github/workflows/ci.yaml:189]:
```suggestion
          cosign upload blob -f test-data/sbom.spdx ${{ matrix.repo }}/sboms@${{ needs.release.outputs.container_digest }}
```

**Review by marcofranssen — COMMENTED** [review]
Looks good although I do think we should first check how much of this can be handled by goreleaser. Please do not remove the branch yet as we can use it for other types of projects that do not use goreleaser.

**Comment by Brend-Smits** [comment]
Closed in favor of #134

**Pull request closed without merging**